api/views.py

```diff
@@ -188,33 +188,41 @@ def confirm_magic_link(request):
 @csrf_exempt
 def send_access_code(request):
     """
-    API endpoint sends a magic link to the user
+    API endpoint confirms presence of study participant matching phone number or device serial number
+    return token if present
     """
     # 1. Parse request parameters.
     data = loadJson(request.body.decode("utf-8"))
-    phone_number = data['phone_number']
-
+    phone_number = data['phone_number'] if 'phone_number' in data.keys() else None
```

Comment:
I think we should break out this out into a separate API endpoint. In general we want to API endpoint name to encapsulate what it is doing. The ability to validate a serial code is hidden.

Thinking we need something like this.

`POST /validate_serial_number`